File: my-app-backend/entity/article.go
package entity

import "time"

type Article struct {
	Id          string    `bson:"_id" json:"article_id"`
	Name        string    `bson:"name" json:"article_name"`
	Author      string    `bson:"author" json:"author"`
	Summary     string    `bson:"summary" json:"summary"`
	CoverImg    string    `bson:"cover_img" json:"cover_img"`
	Catalog     Catalog   `bson:"catalog" json:"catalog"`
	Content     string    `bson:"content" json:"content"`
	Kind        Kind      `bson:"kind" json:"kind"`
	TagList     []Tag     `bson:"tag_list" json:"tag_list"`
	ReleaseTime time.Time `bson:"release_time" json:"release_time"`
	Visibility  int       `bson:"visibility" json:"visibility"`
	Available   bool      `bson:"available" json:"-"`
	CreatedTime time.Time `bson:"created_time" json:"-"`
	UpdatedTime time.Time `bson:"updated_time" json:"-"`
}

type Tag struct {
	Id          string    `bson:"_id" json:"tag_id"`
	Name        string    `bson:"name" json:"tag_name"`
	Available   bool      `bson:"available" json:"-"`
	CreatedTime time.Time `bson:"created_time" json:"-"`
	UpdatedTime time.Time `bson:"updated_time" json:"-"`
}

type Kind struct {
	Id          string    `bson:"_id" json:"kind_id"`
	Name        string    `bson:"name" json:"kind_name"`
	Available   bool      `bson:"available" json:"-"`
	CreatedTime time.Time `bson:"created_time" json:"-"`
	UpdatedTime time.Time `bson:"updated_time" json:"-"`
}

type Catalog struct {
	Value string    `bson:"value" json:"value"`
	Child []Catalog `bson:"child" json:"child"`
}

const (
	_ = iota
	VisibilityPrivate
	VisibilityPublic
	VisibilityDraft
)

File: my-app-backend_r/src/entity/article.rs
use mongodb::bson::DateTime;
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Article {
    #[serde(rename="_id")]
    pub id: String,
    pub name: String,
    pub author: String,
    pub summary: String,
    pub cover_img: String,
    pub catalog: Catalog,
    pub content: String,
    pub kind: Kind,
    pub tag_list: Vec<Tag>,
    pub release_time: DateTime,
    pub visibility: i32,
    pub fulltext_title: String,
    pub fulltext_content: String,
    pub available: bool,
    pub created_time: DateTime,
    pub updated_time: DateTime,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Catalog {
    name: String,
    children: Vec<Catalog>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Kind {
    #[serde(rename="_id")]
    id: String,
    name: String,
    available: bool,
    created_time: DateTime,
    updated_time: DateTime,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Tag {
    #[serde(rename="_id")]
    id: String,
    name: String,
    available: bool,
    created_time: DateTime,
    updated_time: DateTime,
}